### ISSUE TITLE
NullPointerException in org.eclipse.ajdt.internal.ui.editor.quickfix.QuickFixProcessor

### ISSUE THREAD

**Issue by hbrands** [eclipse-aspectj/ajdt#47]
This is 
Spring Tool Suite
eclipse.buildId=4.20.0.202309081654
java.version=17.0.8.1
java.vendor=Eclipse Adoptium
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=de_DE

with 
  AspectJ Compiler	1.9.20.202309080255
  AspectJ Development Tools	2.2.4.202309080255

I noticed this exception in the Eclipse error log, when hovering over an error marker in a java source file with the mouse:

```
java.lang.NullPointerException: Cannot invoke "org.eclipse.ui.IWorkbenchWindow.getActivePage()" because the return value of "org.eclipse.ui.IWorkbench.getActiveWorkbenchWindow()" is null
	at org.eclipse.ajdt.internal.ui.editor.quickfix.QuickFixProcessor.getCorrections(QuickFixProcessor.java:233)
	at org.eclipse.jdt.internal.ui.text.correction.JavaCorrectionProcessor$SafeCorrectionCollector.safeRun(JavaCorrectionProcessor.java:381)
	at org.eclipse.jdt.internal.ui.text.correction.JavaCorrectionProcessor$SafeCorrectionProcessorAccess.run(JavaCorrectionProcessor.java:341)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.jdt.internal.ui.text.correction.JavaCorrectionProcessor$SafeCorrectionProcessorAccess.process(JavaCorrectionProcessor.java:336)
	at org.eclipse.jdt.internal.ui.text.correction.JavaCorrectionProcessor.collectCorrections(JavaCorrectionProcessor.java:465)
	at org.eclipse.jdt.internal.ui.text.java.hover.ProblemHover$ProblemInfo.getJavaAnnotationFixes(ProblemHover.java:115)
	at org.eclipse.jdt.internal.ui.text.java.hover.ProblemHover$ProblemInfo.computeCompletionProposals(ProblemHover.java:89)
	at org.eclipse.jdt.internal.ui.text.java.hover.ProblemHover$ProblemInfo.<init>(ProblemHover.java:74)
	at org.eclipse.jdt.internal.ui.text.java.hover.ProblemHover.createAnnotationInfo(ProblemHover.java:178)
	at org.eclipse.jdt.internal.ui.text.java.hover.AbstractAnnotationHover.getHoverInfo2(AbstractAnnotationHover.java:755)
	at org.eclipse.jdt.internal.ui.text.java.hover.BestMatchHover.getHoverInfo2(BestMatchHover.java:163)
	at org.eclipse.jdt.internal.ui.text.java.hover.BestMatchHover.getHoverInfo2(BestMatchHover.java:130)
	at org.eclipse.jdt.internal.ui.text.java.hover.JavaEditorTextHoverProxy.getHoverInfo2(JavaEditorTextHoverProxy.java:89)
	at org.eclipse.jface.text.TextViewerHoverManager$1.run(TextViewerHoverManager.java:155)
```

(Interestingly, there are errors shown in the java source file but not in the problems view. Not sure of that matters)

**Comment by kriegaex** [comment]
@hbrands Hi Holger! Thanks, I can reproduce this behaviour. Unfortunately, AJDT has not had a propert maintainer for many years, I am merely trying to keep it alive. I know way too little - next to nothing, actually - about OSGi and GUI programming in general and Eclipse plugin development in particular to really have much impact here. Sadly, there are many places which need fixing or rework in AJDT. I can try t find out if it is a trivial fix. Otherwise, I might decide to spend my OSS development time budget otherwise.

As for the problem view: Normally, if you have an error in your source code, the problem view will display if after you save the file, not while still editing it, if the error has been newly introduced.

**Comment by kriegaex** [comment]
The problem should be fixed for the Eclipse 4.23 (2022-03) and 4.26 (2022-12) update sites, also available on the Eclipse Marketplace. Your Eclipse version seems to be 4.20. I am not going to backport, so maybe you want to upgrade. I tested this on 2023-09, and it looks good.

**Comment by hbrands** [comment]
Hi Alexander @kriegaex,
I can confirm this issue is fixed while trying again with the 4.26 update site and Eclipse 2023-12 (Spring STS 4.21.0).
Thanks for your help!

**Comment by kriegaex** [comment]
Actually, for 2023-12 there is a new update site, see [IDE guide](https://github.com/eclipse-aspectj/aspectj/blob/master/docs/developer/IDE.md#aspectj-development-tools-ajdt).
